scipy/stats/tests/test_distributions.py

```diff
@@ -4315,6 +4318,99 @@ def test_burr_nan_mean_var_9544(self):
         assert_(np.isfinite(e4))
 
 
+class TestStudentizedRange:
+    # Critical Q values were selected from this table:
+    # https://link.springer.com/chapter/10.1007/978-1-4613-9629-1_12
+    # For alpha=.05 and .01, and for each value of v=[1, 3, 10, 20],
+    # a Q was picked from each table for k=[2, 8, 14, 20].
+    ks = [2, 8, 14, 20]
+    vs = [1, 3, 10, 20]
```

Comment:
```suggestion
    vs = [1, 3, 10, 20, 120, np.inf]
```

Reply:
I added 120, but np.inf seems to just cause errors with `ppf`:
```
stats.studentized_range.ppf(.95, 8, np.inf)
Traceback (most recent call last):

  File "<ipython-input-80-ced8f501a28f>", line 1, in <module>
    stats.studentized_range.ppf(.95, 8, np.inf)

  File "/Users/swallan/Documents/SciPy/scipy/scipy/stats/_distn_infrastructure.py", line 2140, in ppf
    place(output, cond, self._ppf(*goodargs) * scale + loc)

  File "/Users/swallan/Documents/SciPy/scipy/scipy/stats/_distn_infrastructure.py", line 1017, in _ppf
    return self._ppfvec(q, *args)

  File "/Applications/anaconda3/envs/scipydev/lib/python3.8/site-packages/numpy/lib/function_base.py", line 2113, in __call__
    return self._vectorize_call(func=func, args=vargs)

  File "/Applications/anaconda3/envs/scipydev/lib/python3.8/site-packages/numpy/lib/function_base.py", line 2197, in _vectorize_call
    outputs = ufunc(*inputs)

  File "/Users/swallan/Documents/SciPy/scipy/scipy/stats/_distn_infrastructure.py", line 1807, in _ppf_single
    return optimize.brentq(self._ppf_to_solve,

  File "/Users/swallan/Documents/SciPy/scipy/scipy/optimize/zeros.py", line 776, in brentq
    r = _zeros._brentq(f, a, b, xtol, rtol, maxiter, args, full_output, disp)

RuntimeError: Failed to converge after 100 iterations.
```

and give bad results with `cdf`:
```
1 - stats.studentized_range.cdf(4.286, 8, np.inf)
Out[95]: 0.9960940722046933
```

I suggest that we remove (or fix) the 'switch' at \nu = 10,000, since this the behavior: (alpha=.05 is the expected result.)
@DominicChm thoughts?
```
1 - stats.studentized_range.cdf(4.286, 8, 100000)
Out[99]: 0.9960940722046933

1 - stats.studentized_range.cdf(4.286, 8, 100000-1)
Out[100]: 0.050041523974161284
```

Reply:
Interesting, seems like the asymptotic implementation is broken (100,000 is where it switches in most recent versions). I need to look into this. 

A similar case on my graphs shows that the regular integral is able to get out to v~=300,000 before starting to fail. 

![k-8](https://user-images.githubusercontent.com/45153623/112375702-cef8cd00-8ca0-11eb-85ab-c878fbdd983e.png)